### ISSUE TITLE
[Feature Request] Compatible with the latest version of Vuepress

### ISSUE THREAD

**Issue by Zhengqbbb** [Renovamen/vuepress-theme-gungnir#61]
### 💭 Describe the feature

Unfortunately, vuepress-next has recently undergone a huge refactoring. I just looked at the demotion and it doesn't work properly. Both the type and the plugin are written differently😫

**Comment by Renovamen** [comment]
Yeah, there are a lot of breaking changes in VuePress [v2.0.0-beta.40](https://github.com/vuepress/vuepress-next/blob/main/CHANGELOG.md#200-beta40-2022-04-25)...  🤔

I'll try to migrate to the latest version, but it may take some time since I'm  recently busy with homework, final projects and exams. Before that, please stick to VuePress v2.0.0-beta.39. I'll also add it to the documentation. Thanks!

**Comment by Zhengqbbb** [comment]
> Please stick to VuePress v2.0.0-beta.39. I'll also add it to the documentation. Thanks!

Pin the vuepress-version: **v2.0.0-beta.39** can work !
@Renovamen 
I found that vuepress still has a lot of design changes to be made, so it is better not to be compatible recently.

---
version-list:
```
npmPackages:
    @vuepress/bundler-vite:  2.0.0-beta.39
    @vuepress/bundler-webpack:  2.0.0-beta.38
    @vuepress/cli:  2.0.0-beta.38
    @vuepress/client:  2.0.0-beta.38
    @vuepress/core:  2.0.0-beta.38
    @vuepress/markdown:  2.0.0-beta.38
    @vuepress/plugin-active-header-links:  2.0.0-beta.38
    @vuepress/plugin-back-to-top:  2.0.0-beta.38
    @vuepress/plugin-container:  2.0.0-beta.38
    @vuepress/plugin-docsearch: 2.0.0-beta.38 => 2.0.0-beta.38
    @vuepress/plugin-external-link-icon:  2.0.0-beta.38
    @vuepress/plugin-git:  2.0.0-beta.38
    @vuepress/plugin-google-analytics:  2.0.0-beta.38
    @vuepress/plugin-medium-zoom:  2.0.0-beta.38
    @vuepress/plugin-nprogress:  2.0.0-beta.38
    @vuepress/plugin-palette:  2.0.0-beta.38
    @vuepress/plugin-prismjs:  2.0.0-beta.38
    @vuepress/plugin-pwa:  2.0.0-beta.38
    @vuepress/plugin-pwa-popup:  2.0.0-beta.38
    @vuepress/plugin-register-components: Not Found
    @vuepress/plugin-search: Not Found
    @vuepress/plugin-shiki: Not Found
    @vuepress/plugin-theme-data:  2.0.0-beta.38
    @vuepress/plugin-toc: Not Found
    @vuepress/shared:  2.0.0-beta.38
    @vuepress/theme-default:  2.0.0-beta.38
    @vuepress/utils:  2.0.0-beta.38
    vue:  3.2.33
    vue-loader:  17.0.0
    vue-router:  4.0.14
    vuepress: 2.0.0-beta.39 => 2.0.0-beta.39
    vuepress-vite: 2.0.0-beta.39 => 2.0.0-beta.39
    vuepress-webpack: 2.0.0-beta.38 => 2.0.0-beta.38
```

**Comment by Zhengqbbb** [comment]
@Renovamen 
I also finished compatible with vuepress v2.0.0-beta.45 today. At present, it seems that there is a problem with the build, using vite and there are a lot of vue warning messages, I suspect that vuepress-next should not have completed this work, anyway, it is very nice

**Comment by Renovamen** [comment]
@Zhengqbbb Thanks! I guess the warnings are caused by @vuepress/plugin-pwa-popup (https://github.com/vuepress/vuepress-next/issues/903), which should be fixed in [this commit](https://github.com/vuepress/vuepress-next/commit/3daecdd2a3fa48c734040b88b3491ca11442c260). I'll keep up with the new release.

**Comment by Renovamen** [comment]
@Zhengqbbb The problem has been fixed in the latest theme release (2.0.0-alpha.24), so I'm closing this issue now. Thank you!
